### ISSUE TITLE
Silent Compilation Errors?

### ISSUE THREAD

**Issue by retailcoder** [WaynePhillipsEA/twinbasic#21]
**Describe the bug**

It appears the linker requires a default constructor, but the compiler isn't complaining *and a .dll file is still being output*, which is confusing:

> [LINKER] compilation error detected in 'Test.{default_constructor}'
> [LINKER] FAILED due to compilation errors 'path\to\project\build\library_win32.dll'
> [BUILD] failed

**To Reproduce**
1. Add a class module with a parameterized constructor *but no default parameterless constructor*, fix any compiler errors ("problems") in the project
2. Clear the \build folder
3. Build the project and bring up the _Debug Console_
4. Linker failed due to _silent_ compilation errors, but \build now includes the built .dll

```vb
class Test
    sub New(byval a as string)
    end sub
end class
```

**Expected behavior**

Compilation errors shouldn't be silent, and if any debug output is reporting a failed build, there probably shouldn't be a .dll in the build output folder.

**Comment by WaynePhillipsEA** [comment]
I agree, for an exposed class the compiler should verify that there is a (COM compatible) default constructor at compile time, not link time.  OR we should allow a default constructor to be auto-generated, despite there being only an explicit parameterized constructor.

**Comment by WaynePhillipsEA** [comment]
Fixed in v0.9.1622
